File: greed_is_good_js/score.js
function score(dice) {
  let points = 0

  for (let i = 1; i < 7; i++) {
    const count = dice.filter(x => x === i).length
    if (i === 1) {
      points += (count >= 3 ? 1000 : 0) + (100 * (count % 3))
    } else {
      points += count >= 3 ? 100 * i : 0
    }

    if (i === 5) {
      points += 50 * (count % 3)
    }
  }

  return points
}
module.exports = score;

File: python/greed_is_good/score.py
def score(dice):
  points = 0
  for x in range(1, 7):
    count = dice.count(x)
    if x == 1:
      points += (1000 if count >= 3 else 0) + (100 * (count % 3))
    else:
      points += 100 * x if count >= 3 else 0
    if x == 5:
      points += 50 * (count % 3)

  return points
